analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Nullable.cs

```diff
@@ -85,4 +85,18 @@ public void Nullable_HasValue_ReadsBoolConstraintFromObjectConstraint()
         validator.ValidateTag("HasValueAfterNull", x => x.HasConstraint(BoolConstraint.False).Should().BeTrue());
         validator.ValidateTag("SymbolAfterNull", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
     }
+
+    [TestMethod]
+    public void Nullable_Ctor_NoArguments_SetsNullConstraint()
+    {
+        const string code = """
+            int? explicitType = new Nullable<int>();
+            int? targetTyped = new();
+            Tag("ExplicitType", explicitType);
+            Tag("TargetTyped", targetTyped);
+            """;
+        var validator = SETestContext.CreateCS(code).Validator;
+        validator.ValidateTag("ExplicitType", x => x.HasConstraint(ObjectConstraint.Null).Should().BeTrue());
+        validator.ValidateTag("TargetTyped", x => x.HasConstraint(ObjectConstraint.NotNull).Should().BeTrue("new() of int produces value 0"));
```

Comment:
😮 Interesting behavior.

Reply:
Indeed. It took me a debug session and a manual test to verify. definitely deserves a special message :)